src/gennylib/test/PoolFactory_test.cpp

```diff
@@ -237,6 +237,16 @@ TEST_CASE("PoolFactory behavior") {
 
         auto pool = factory.makePool();
         REQUIRE(pool);
+
+        // We should be able to change the value of an option
+        auto expectedRedactedUri = [&]() {
```

Comment:
Given my other comment, don't update this test, I think this is a great way of testing it.

Reply:
As I'm gonna be creating the redacted DSN copy from the actual DSN instead of calling makeUri(), do you think I should keep my changes in this file or shouldn't?

Reply:
Leave this how it is. I think testing using a different method from generating it originally is a great approach. Imagine if you're using `strings.sub(password, "REDACTED")` in the main function AND the test function, and for some reason `strings.sub` breaks and does nothing, the test wont pick it up.

What this test is currently stating is "the redacted DSN should look exactly as if we generated a DSN with `redacted` as a password, which is bang on.

Reply:
We've discussed this offline, and we agree that it's not worth setting up mocking just to test the log. The patch build shows that the code works anyway.